### ISSUE TITLE
Broken After Latest Discord Update

### ISSUE THREAD

**Issue by QHV154** [EnKrypt/DiscordEncryption#4]
Encrypted Messages can be read but new encrypted messages can't be sent. If you type a message with the padlock activated it sends the message but not encrypted. Sometimes having the plugin active can cause loss of ability to type in the message box. These issues are consistent with all I know who use this plugin.

**Comment by EnKrypt** [comment]
Yup, you're right. I just noticed this today. Thanks for reporting.

Give me a while. I'll spend some time debugging and update this issue when I have more insight.

**Comment by EnKrypt** [comment]
I have some bad news. 

In the last update, Discord seems to have completely changed how the message textbox works and replaced it with an instance of some [slate editor](https://github.com/ianstormtaylor/slate).

Sadly this makes things quite complicated. It's a breaking change of course, but it also implies that half of the entire plugin might need to be re-written.

Having already spent an hour to debug, I simply don't have the time at this moment to dedicate the half a dozen or so hours it would take to actually fix this.

So hopefully at some point in the future, I will get back to this and this plugin will work again (till then, I advise keeping this plugin disabled).

If you're reading this and you're sad about it, there are a few things you could do to help :
* Attempt to fix it yourself. All kinds of PR are welcome of course. I will take the time to participate in discussions and help you even if I may not have the time to code myself.
* Find out how to get a hold of the Discord slate editor instance within React.
* Find out how to access the slate editor APIs to do things such as getting values, setting values, overriding event handlers, etc.)
* If you don't know how to do the above things, find someone in your network using the plugin who is a programmer and does.

**Comment by EnKrypt** [comment]
Adding the hacktoberfest label in case someone wants to have a look at this.
Be warned, this is likely not a simple task to fix. I did some preliminary investigation in my free time and found that the underlying modules Discord is using for capturing and queuing messages isn't straightforward.

That said, if you enjoy reverse engineering this is something you can have a lot of fun with.
